src/array/cpu/spmm.h

```diff
@@ -267,10 +267,10 @@ void SpMMCmpCsr(
 #if !defined(_WIN32)
 #ifdef USE_LIBXSMM
   int cpu_id = libxsmm_cpuid_x86();
-  const bool no_libxsmm =
-      bcast.use_bcast || std::is_same<DType, double>::value ||
-      (std::is_same<DType, BFloat16>::value && cpu_id < LIBXSMM_X86_AVX512) ||
```

Comment:
I see you removed BF16 check `std::is_same<DType, BFloat16>::value`. Is it an overlap with / equivalent to `cpu_id < LIBXSMM_X86_AVX512)`? Would it reopen this issue https://github.com/dmlc/dgl/issues/5647 on some types of CPU?

Reply:
Previously this path was broken for bf16 data type and avx2, now it is broken for all data types.
That's why I modified this line, to disable `SpMMCmpCsrLibxsmm` for platforms without AVX512

Reply:
I see. To confirm, SpMMCmpCsrLibxsmm works with BF16 with AVX512 on all platforms?

Reply:
`SpMMCmpCsrLibxsmm` for all data types works only for AVX512 platforms